pkg/controllermanager/controller/project/activity/add.go

```diff
@@ -0,0 +1,167 @@
+	// the controller manually.
+	c, err := controller.New(
+		ControllerName,
+		mgr,
+		controller.Options{
+			Reconciler:              r,
+			MaxConcurrentReconciles: pointer.IntDeref(r.Config.ConcurrentSyncs, 0),
+			RecoverPanic:            true,
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	if err := c.Watch(
+		&source.Kind{Type: &gardencorev1beta1.Shoot{}},
+		mapper.EnqueueRequestsFrom(mapper.MapFunc(r.MapObjectToProject), mapper.UpdateWithNew, c.GetLogger()),
+		r.OnlyNewlyCreatedObjects(),
+		predicate.GenerationChangedPredicate{},
+	); err != nil {
+		return err
+	}
+
+	if err := c.Watch(
+		&source.Kind{Type: &gardencorev1beta1.BackupEntry{}},
+		mapper.EnqueueRequestsFrom(mapper.MapFunc(r.MapObjectToProject), mapper.UpdateWithNew, c.GetLogger()),
+		r.OnlyNewlyCreatedObjects(),
+		predicate.GenerationChangedPredicate{},
+	); err != nil {
+		return err
+	}
+
+	if err := c.Watch(
+		&source.Kind{Type: &gardencorev1beta1.Quota{}},
+		mapper.EnqueueRequestsFrom(mapper.MapFunc(r.MapObjectToProject), mapper.UpdateWithNew, c.GetLogger()),
+		r.OnlyNewlyCreatedObjects(),
+		r.NeedsSecretBindingReferenceLabelPredicate(),
+	); err != nil {
+		return err
+	}
+
+	return c.Watch(
+		&source.Kind{Type: &corev1.Secret{}},
+		mapper.EnqueueRequestsFrom(mapper.MapFunc(r.MapObjectToProject), mapper.UpdateWithNew, c.GetLogger()),
+		r.OnlyNewlyCreatedObjects(),
+		r.NeedsSecretBindingReferenceLabelPredicate(),
+	)
+}
+
+// OnlyNewlyCreatedObjects filters for objects which are created less than an hour ago for create events. This can be
+// used to prevent unnecessary reconciliations in case of controller restarts.
+func (r *Reconciler) OnlyNewlyCreatedObjects() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			objMeta, err := meta.Accessor(e.Object)
+			if err != nil {
+				return false
+			}
+
+			return r.Clock.Now().UTC().Sub(objMeta.GetCreationTimestamp().UTC()) <= time.Hour
+		},
+	}
+}
+
+// NeedsSecretBindingReferenceLabelPredicate returns a predicate which only returns true when the objects have the
+// reference.gardener.cloud/secretbinding label.
+func (r *Reconciler) NeedsSecretBindingReferenceLabelPredicate() predicate.Predicate {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			objMeta, err := meta.Accessor(e.Object)
+			if err != nil {
+				return false
+			}
+
+			_, hasLabel := objMeta.GetLabels()[v1beta1constants.LabelSecretBindingReference]
+			return hasLabel
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldObjMeta, err := meta.Accessor(e.ObjectOld)
+			if err != nil {
+				return false
+			}
+
+			objMeta, err := meta.Accessor(e.ObjectNew)
+			if err != nil {
+				return false
+			}
+
+			_, oldObjHasLabel := oldObjMeta.GetLabels()[v1beta1constants.LabelSecretBindingReference]
+			_, newObjHasLabel := objMeta.GetLabels()[v1beta1constants.LabelSecretBindingReference]
+
+			return oldObjHasLabel && newObjHasLabel
```

Comment:
Previously, it was like this,
https://github.com/gardener/gardener/blob/958fec3e428d6bcc9b115b0423fd5ba51981b06d/pkg/controllermanager/controller/project/project_activity_control.go#L103-L123
which is not the case now.
On simple words, we were acting if either of the object has the label. If the new doesn't have the label and the old had, which means the deletion of the last secretbinding referring this secret. Don't we want to consider this as "activity"?
If not, then we should only consider the newObject. The first time the secret is "referred" the oldObject won't have label, but we want to consider this as activity too.

Reply:
I don't remember to have changed this on purpose, so probably it was a mistake. @shafeeqes can you open a PR to fix it and preserve the old behaviour?

Reply:
Sure, but should we consider the deletion of the last `secretbinding` (which can be in a different namespace) as an activity?

Reply:
Yes, let's consider it as activity 👍🏻

Reply:
Okay, then I'll go ahead.